openedx/core/lib/cache_utils.py

```diff
@@ -4,42 +4,105 @@
+    the duration of the request. It returns the cached value for subsequent
+    calls to the same function, with the same parameters, within a given request.
+
+    Notes:
+        - We convert arguments and keyword arguments to their string form to build the cache key. So if you have
+          args/kwargs that can't be converted to strings, you're gonna have a bad time (don't do it).
+        - Cache key cardinality depends on the args/kwargs. So if you're caching a function that takes five arguments,
+          you might have deceptively low cache efficiency.  Prefer functions with fewer arguments.
+        - WATCH OUT: Don't use this decorator for instance methods that take in a "self" argument that changes each
+          time the method is called. This will result in constant cache misses and not provide the performance benefit
+          you are looking for. Rather, change your instance method to a class method.
+        - Benchmark, benchmark, benchmark! If you never measure, how will you know you've improved? or regressed?
 
     Arguments:
-        request_cache_attr_name - The name of the field or property in this method's containing
-         class that stores the request_cache.
+        namespace (string): An optional namespace to use for the cache. By default, we use the default request cache,
+            not a namespaced request cache. Since the code automatically creates a unique cache key with the module and 
+            function's name, storing the cached value in the default cache, you won't usually need to specify a
+            namespace value.
+            But you can specify a namespace value here if you need to use your own namespaced cache - for example,
+            if you want to clear out your own cache by calling RequestCache(namespace=NAMESPACE).clear().
+        arg_map_function (function: arg->string): Function to use for mapping the wrapped function's arguments to
+            strings to use in the cache key. If not provided, defaults to force_text, which converts the given
+            argument to a string.
+        request_cache_getter (function: args, kwargs->RequestCache): Function that returns the RequestCache to use.
+            If not provided, defaults to edx_django_utils.cache.RequestCache.  If None, the function's return values
+            are not cached.
+
+    Returns:
+        func: a wrapper function which will call the wrapped function, passing in the same args/kwargs,
+              cache the value it returns, and return that cached value for subsequent calls with the
+              same args/kwargs within a single request.
     """
-    def _decorator(func):
-        """Outer method decorator."""
-        @functools.wraps(func)
-        def _wrapper(self, *args, **kwargs):
+    def decorator(f):
+        """
+        Arguments:
+            f (func): the function to wrap
+        """
+        # @functools.wraps(f)
```

Comment:
Going back on this -- without the `functools.wraps` call, I believe if you have a function `my_real_fn_name()` that you decorate with `@request_cached()`, and then later on if you're in Python and have a reference to `my_real_fn_name` and ask `my_real_fn_name.__name__` or `my_real_fn_name.__doc__`, you're going to get `_decorator` and the docstring for `_decorator` instead of the values for `my_real_fn_name`. The `functools.wraps` does the bookkeeping so that decorators don't wipe out that information from the functions they wrap.

Reply:
Oh, yes.  For sure.  I think I commented this out when I was debugging something else.  @robrap If you're inheriting this PR, please remember to uncomment this.